helix-core/src/main/java/org/apache/helix/controller/stages/IntermediateStateCalcStage.java

```diff
@@ -709,17 +698,17 @@ private void logPartitionMapState(String resource, Set<Partition> allPartitions,
   /**
    * POJO that maps resource name to its priority represented by an integer.
    */
-  private static class ResourcePriority {
-    private String _resourceName;
-    private int _priority;
+  private static class ResourcePriority implements Comparable<ResourcePriority> {
+    private final String _resourceName;
+    private int _priority = Integer.MIN_VALUE;
 
-    ResourcePriority(String resourceName, Integer priority) {
+    ResourcePriority(String resourceName) {
       _resourceName = resourceName;
-      _priority = priority;
     }
 
+    @Override
     public int compareTo(ResourcePriority resourcePriority) {
-      return Integer.compare(_priority, resourcePriority._priority);
+      return Integer.compare(resourcePriority._priority, _priority);
```

Comment:
curious, why we change order?

Reply:
In the old implementation, the comparator used to call this compareTo the reverse way